data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/ScanObjectWorker.java

```diff
@@ -91,19 +127,53 @@ private void startProcessingObject(final int waitTimeMillis) {
         final String objectKey = objectToProcess.get().getPartitionKey().split("\\|")[1];
 
         try {
-            processS3Object(S3ObjectReference.bucketAndKey(bucket, objectKey).build());
-            sourceCoordinator.completePartition(objectToProcess.get().getPartitionKey());
+            List<DeleteObjectRequest> waitingForAcknowledgements = new ArrayList<>();
+            AcknowledgementSet acknowledgementSet;
+            CompletableFuture<Boolean> completableFuture = new CompletableFuture<>();
+
+            acknowledgementSet = acknowledgementSetManager.create((result) -> {
+                acknowledgementSetCallbackCounter.increment();
+                // Delete only if this is positive acknowledgement
+                if (result == true) {
```

Comment:
Yes, but it's not useful as the callback is already called and it won't timeout. We just don't close the partition or delete S3 object in case of `false`.